### DISCUSSION TITLE
Setting up an 1136 to measure air pressure in Artisan - help needed

### DISCUSSION THREAD

**Discussion by lukehawkins23** [artisan-roaster-scope/artisan#807] [Q&A]
I have an 1136 attached to a VINT Hub. I also run 2 TCs off the VINT Hub using an 1101. I'm having no trouble with the TCs, but I cannot get the 1136 to show a kpa value in roast scope. I am using the 1136 as an additional device, plugged into Port 2 on the VINT Hub. I go into the Phidgets control panel and select the 1136 through the voltage ratio mode, and the control panel shows the voltage reading and correctly converts it to a kpa value.

Artisan, is not showing the kpa values. Please see attached file for the extra device settings. They are probably not right, but I have played around with them a lot and not much changes. Clearly something is wrong in the settings. The LCD always shows a 0.5 reading which won't change, the curve LCD shows 1.0 which also doesn't change. 

I also tried playing with the settings in the red box of this image, but nothing happened.

![image](https://user-images.githubusercontent.com/98791782/151947771-913643d8-b76e-498b-a094-e85cf69df11e.png)

Any help would be massively appreciated as I have spent around 6 hours trying to figure this out. Happy to provide any additional information anyone needs. 

Luke
![image](https://user-images.githubusercontent.com/98791782/151947325-54583af3-de4a-47f2-82ac-068a028a39b2.png)

**Comment by MAKOMO** [comment]
Could you please attach your artisan-settings.aset file (save via menu Help >> Save Settings and rename into artisan-settings.txt to drag-and-drop here to GitHub)? Which HUB port is the TMP1101 connected?

**Reply by lukehawkins23** [reply]
The 1101 is connected via Port 1. Hope the settings file works, I had to change the file name after saving. 
[artisan-settings.txt](https://github.com/artisan-roaster-scope/artisan/files/7978094/artisan-settings.txt)
.

**Reply by MAKOMO** [reply]
I suggest to connect the 1136 into HUB port 0 and use the "Phidget HUB IO 0" device type to read it following the excellent guide of Michael linked below. That way only one HUB port is blocked. If you use "Phidget HUB IO 23", both, the port 2 and 3 will be blocked even if your using only port 2.

You should use the formula `x-T3` for y1(x) as the first extra device channel 1 is linked to T3 and not T4 (which would be the second channel of the first extra device).

The tarred reading should be referred to by `Y3` (and not `Y4`) in your second (Virtual) extra device, channel 1 symbolic function y1(x).

**Comment by CarefreeBuzzBuzz** [comment]
Take a look at the doc here for some settings and leads that may be useful.

https://www.carefreebuzzbuzz.com/Learning-About-Coffee/mill-city-roaster-upgrades
----------------------------------------
Michael Herbert, Head Roaster
Order at www.carefreebuzzbuzz.com <http://www.carefreebuzzbuzz.com/>
Artisan RoasterScope  <https://artisan-scope.org/>Team

**Reply by lukehawkins23** [reply]
<img width="107" alt="Screen Shot 2022-02-02 at 6 28 24 pm" src="https://user-images.githubusercontent.com/98791782/152111629-4f2bf1b1-b315-4243-bd2a-e4f00b7dff6f.png">
Doing that I get values that look like this. They do change with the pressure changes, but the values are neither voltage readings or kpa readings.

**Reply by lukehawkins23** [reply]
There is a chance that these values are somehow related to kpa, but I currently don't have a reliable way to tell. When I have artisan 'On', I can't view the kpa and voltage values in the phidgets control panel because it says it's being called by another program (artisan). I do have a manometer attached to the roaster which I can correlate with, but the values aren't matching in ratio or actual value. 
I also don't quite understand the tare function. Right clicking gives me all kinds of values both positive and negative. And they just keep changing with seemingly no basis for what 'tare' actually is.

**Reply by lukehawkins23** [reply]
<img width="93" alt="Screen Shot 2022-02-02 at 6 54 14 pm" src="https://user-images.githubusercontent.com/98791782/152114635-9d3b9088-3b56-4e49-bf29-d965ea8cd2f7.png">
With a bit of playing I get a value on the LCD like in this screenshot. My manometer shows 0.2kpa and there's a chance this value is changing in line with the pressure changes, but just show 171.x instead of 0.171kpa. I need to try graphing by adding a formula that divides it by 1000 to see if maybe it is working properly. Ideally though, in order to graph the value on the roast scope in a useful way I also need to make 100 degreesC the 2.0kpa point and 0 degrees the 0.0kpa, if that makes sense.

**Reply by lukehawkins23** [reply]
Sorry, lots of replies!
I'm not sure it's working properly in Artisan. In the phidgets control panel the 1136 is incredibly sensitive to pressure changes and matches the manometer closely. In artisan it's not very sensitive at all, especially at lower pressures and it's readings just aren't corresponding.

**Reply by bigl00z3** [reply]
I don't if this can help but just in case...

When looking at the [1136 page](https://www.phidgets.com/?tier=3&catid=7&pcid=5&prodid=109), in the user guide section, there is the formula to convert the voltage ratio value to kPa or psi:

_Formulas
The Phidget libraries can automatically convert sensor voltage into pressure (kPa) by selecting the appropriate SensorType. See the Phidget22 API for more details. The Formula to translate voltage ratio into pressure is:

`Pressure (kPa)=(5×VoltageRatio)−2.5`

`Pressure (psi)=(0.725×VoltageRatio)−0.3625`_

Nico

**Reply by lukehawkins23** [reply]
So theoretically then, could I have the 1136 just as a voltage meter in the phidgets control panel, then in Artisan use the formula for graphic the 1136 Y=(5xX)-2.5 to take the voltage and graph as kpa?

**Reply by bigl00z3** [reply]
That's actually what the 1136 does. If you look at the Specifications and User manual parts of the [1136 page](https://www.phidgets.com/?tier=3&catid=7&pcid=5&prodid=109)  : 

> _[1136 - Differential Air Pressure Sensor ±2 kPa](https://www.phidgets.com/products.php?product_id=1136)
> **Any Phidget with a Voltage Ratio Input port**,_

Some Phidgets are communicating data as a Voltage Ratio Input (such at this one), other Phidgets are communicating the converted data. The second type are market as "objects" for example the [PRE1000](https://www.phidgets.com/?tier=3&catid=7&pcid=5&prodid=719) will be recognise as such.
You can see on the enclosure the OBJECT mention
![image](https://user-images.githubusercontent.com/8886922/153765972-b09af981-0d5a-4c26-baef-dd01aa72efac.png)

You can also see them as objects in the Phidget Control Panel:
![image](https://user-images.githubusercontent.com/8886922/153766066-12bf96d3-0e43-4de2-8dc7-f0399432ce16.png)

For the non Object models, you get the raw data which you have to process yourself (in our case in Artisan). 
When not sure just have a look at the very complete documentation on the Phidget website. 

Nico

**Comment by lukehawkins23** [comment]
I have now got the 1136 working as I'd like it to. I am having trouble getting the 1136 curve to graph in a readable way as events. The curve actually looks quite useful, as can be seen in the first image. But when I add the events, there are too many events showing and it is necessary. The reason is that the gas pressure changes aren't instantaneous, and so it is registering smaller changes on the way to the desired value. I'd like to find a way to remove some of the values (without having to manually delete events) to make it cleaner. Any ideas on how to do this
<img width="1169" alt="Screen Shot 2022-02-16 at 8 14 35 pm" src="https://user-images.githubusercontent.com/98791782/154233639-072f7cae-152c-49fe-b022-680d6a16763d.png">
<img width="1177" alt="Screen Shot 2022-02-16 at 8 14 46 pm" src="https://user-images.githubusercontent.com/98791782/154233649-5a4a4f0b-60d8-4e62-beba-1ea7a4eeaf6c.png">
?

**Reply by MAKOMO** [reply]
The quantification mechanism in Artisan is not intended to be used on PV (Process Values) as you have them here, but on SV (Set Values; Control Signals). By default quantification is by 1/100% steps (1% steps with a range from 0-100%). You can change this to 10/100% steps (10% steps with a range of 0-100%) by ticking the Coarse flag of the slider specification. You will end up with marks only at 0%, 10%,  20%... By "playing" with the min/max range you can also different steps. A coarse slider with a range of 0-50% will have a step width of 5%.. Why not just work with the raw signal or those quantified events that you have just without those additional annotation strings?

**Reply by lukehawkins23** [reply]
Thanks for this, I'll have a play. The main reason for anting the annotation is repeatability and also for getting feedback from a pro. I need to have those values in there for that purpose.

**Reply by lukehawkins23** [reply]
This didn't work as suggested. I could get the course setting to show 10% increment changes, adjusting the maximum didn't change the step widths in percentage, it just changed the range the steps appear on the graph, so going to say 0-50 halved the height of the graph, 0-200 doubled it, but the actual set values remained the same.

**Comment by MAKOMO** [comment]
Are you using a Phidget 1136 to measure gas pressure? Those sensors are not qualified for gas, but only for air! Did you notice the warning on the [product page](https://www.phidgets.com/?tier=3&pcid=5&prodid=111) reading

`This sensor is designed to measure the pressure of clean, non-condensing air. Contact with moist air (i.e, human breath), dusty air, or other gases can damage the sensor.`

I guess the sensor might even leak if used with propane or NG gas.

**Reply by lukehawkins23** [reply]
Hi, yes I know, I intend to replace it with something designed for propane. It's $400 though so want to get everything right first. It doesn't leak gas thankfully, I did test that!

**Comment by Pieterdwhome** [comment]
Spend 2 days trying to get this to work. Good information in the thread but the important missing step is to select the port as a ratiometric sensor by selecting the ratio tick box for the channel you are using. I am using 3 ratiometric sensors hence I have 3 ports ticked with "ratio. Then simply use the formula in the sensor's user manual and it all works. 
<img width="752" alt="Screenshot 2022-06-12 at 09 56 40" src="https://user-images.githubusercontent.com/107343542/173223262-162b1689-372b-4d93-a8de-d9a9191f20aa.png">
"

**Reply by leCoffea** [reply]
HI, is there anyway you could step by step getting the 1136 to work? 
I have the input voltage reading in artisan but I can't seem to get the output voltage ratio to read. I'm not much of a coder. tips?

**Comment by CarefreeBuzzBuzz** [comment]
Try the PDF at this link to see if it gets you started. https://www.carefreebuzzbuzz.com/Learning-About-Coffee/mill-city-roaster-upgradesMichael Herbert CarefreeBuzzBuzz On Feb 9, 2024, at 9:05 PM, leCoffea ***@***.***> wrote:﻿
HI, is there anyway you could step by step getting the 1136 to work?
I have the input voltage reading in artisan but I can't seem to get the output voltage ratio to read. I'm not much of a coder. tips?
